cirq-google/cirq_google/engine/processor_config.py

```diff
@@ -0,0 +1,107 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import cirq_google as cg
+from cirq_google.api import v2
+from cirq_google.engine import util
+
+if TYPE_CHECKING:
+    import cirq
+    from cirq_google.cloud.quantum_v1alpha1.types import quantum
+
+
+class ProcessorConfig:
+    """Representation of a quantum processor configuration
+
+    Describes available qubits, gates, and calivration data associated with
+    a processor configuration.
+
+    Raise:
+        ValueError: If quantum_processor_config is contains incompatible types.
+    """
+
+    def __init__(
+        self, *, quantum_processor_config: quantum.QuantumProcessorConfig, run_name: str = ''
+    ) -> None:
+        self._quantum_processor_config = quantum_processor_config
+        self._run_name = run_name
+        self._grid_device = cg.GridDevice.from_proto(
+            util.unpack_any(
+                self._quantum_processor_config.device_specification,
+                v2.device_pb2.DeviceSpecification(),
+            )
+        )
+        self._calibration = cg.Calibration(
+            util.unpack_any(
+                self._quantum_processor_config.characterization, v2.metrics_pb2.MetricsSnapshot()
+            )
+        )
+
+    @property
+    def effective_device(self) -> cirq.Device:
+        """The GridDevice generated from thisc configuration's device specification"""
+        return self._grid_device
+
+    @property
+    def calibration(self) -> cg.Calibration:
+        """Charicterization metrics captured for this configuration"""
+        return self._calibration
+
+    @property
+    def snapshot_id(self) -> str:
+        """The snapshot that contains this processor config"""
+        if 'configSnapshots' not in self._quantum_processor_config.name:
+            # We assume the calling `get_quantume_processor_config` always
+            # returns a config with the snapshot resouce nanme.  This check
+            # is added in case this behavior changes in the future.
+            return ''
+        parts = self._quantum_processor_config.name.split('/')
+        return parts[5]
+
+    @property
+    def run_name(self) -> str:
+        """The run that generated this config if avaiable."""
+        return self._run_name
+
+    @property
+    def project_id(self) -> str:
+        """The project that contains this config."""
+        parts = self._quantum_processor_config.name.split('/')
+        return parts[1]
+
+    @property
+    def processor_id(self) -> str:
+        """The processor id for this config."""
+        parts = self._quantum_processor_config.name.split('/')
+        return parts[3]
+
+    @property
+    def config_alias(self) -> str:
```

Comment:
We call this "config_name" the Quantum Engine API. @hoisinberg - what do you think about converging on the "config_name" over the older "config_alias" as we update all of this config tooling?